pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -428,8 +427,8 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation) *gardencorev1bet
 		})
 
 		destroyNginxIngressDNSRecord = g.Add(flow.Task{
-			Name:         "Destroying ingress DNS record",
-			Fn:           flow.TaskFn(component.OpDestroyAndWait(botanist.Shoot.Components.Extensions.DNS.NginxEntry).Destroy),
+			Name:         "Destroying nginx ingress DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyIngressDNSRecord),
```

Comment:
Hmm, why `component.OpDestroyAndWait` is now switched to only `component.OpDestroy` (in `DestroyIngressDNSRecord` func)? I guess it now won't wait until the resources are deleted.

Reply:
@mvladev , for 1 more time I want to share that for me the "component" semantic is too messy - you have `Wait` vs `WaitForCleanup`; you can `Deploy` but with `invert=true` which does `Destroy`...

Reply:
Fixed